packages/react/src/components/OverflowMenu/OverflowMenu.tsx

```diff
@@ -369,26 +501,26 @@ class OverflowMenu extends Component {
   };
 
   /**
-   * Focuses the next enabled overflow menu item given the currently focused
-   * item index and direction to move
-   * @param {object} params
-   * @param {number} params.currentIndex - the index of the currently focused
-   * overflow menu item in the list of overflow menu items
-   * @param {number} params.direction - number denoting the direction to move
-   * focus (1 for forwards, -1 for backwards)
+   * Focuses the next enabled overflow menu item given
+   * the currently focused item index and direction to move
    */
-  handleOverflowMenuItemFocus = ({ currentIndex, direction }) => {
-    const enabledIndices = React.Children.toArray(this.props.children).reduce(
-      (acc, curr, i) => {
-        if (!curr.props.disabled) {
-          acc.push(i);
-        }
-        return acc;
-      },
-      []
-    );
+  handleOverflowMenuItemFocus = ({
+    currentIndex,
+    direction,
+  }: FocusHandlerOptions) => {
+    const enabledIndices: number[] = [];
+    React.Children.forEach<ChildElement>(this.props.children, (child, i) => {
+      if (
+        isElement(child) &&
+        child.type === OverflowMenuItem &&
+        !child.props.disabled
+      ) {
+        enabledIndices.push(i);
+      }
+    });
```

Comment:
Same thing here - we can't require that the children be OverflowMenuItem. Why is the additional check for `isElement` necessary?

Reply:
Because otherwise the `child` variable is an expansive type including crap like `boolean`, `string`, `number` and whatnot… the `isElement` narrows the type to `ReactElement`